detection_rules/main.py

```diff
@@ -179,6 +182,24 @@ def import_rules_into_repo(input_file: click.Path, required_only: bool, action_c
         if isinstance(contents["author"], str):
             contents["author"] = [contents["author"]]
 
+        local_metadata = get_rule_metadata_from_file(Path(rule_path))
+        if local_metadata:
+            contents["maturity"] = local_metadata.get("maturity", "development")
+            contents.update(
+                {
+                    **(
+                        {"creation_date": local_metadata.creation_date}
+                        if local_creation_date and local_metadata.creation_date
+                        else {}
+                    ),
+                    **(
+                        {"updated_date": local_metadata.updated_date}
+                        if local_updated_date and local_metadata.updated_date
+                        else {}
+                    ),
+                }
+            )
```

Comment:
Looks similar to the logic in kbwrap. can we turn it into a function and call it in both places?

Reply:
Sounds great will do!

Reply:
Updated